session_3/blogspot/app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,86 @@
+class ArticlesController < ApplicationController
+  load_and_authorize_resource
+
+  before_action :set_article, only: [:show, :edit, :update, :destroy]
+
+  def article_params
+    params.require(:article).permit(:title, :topic, :tags, :content, :public)
+  end
+
+  # GET /articles or /articles.json
+  def index
+    @articles = Article.all
+    if !current_user
+      session[:private_access] = 5
+    end
+  end
+
+  # GET /articles/1 or /articles/1.json
+  def show
+    if current_user
+     if can? :update, @article, user_id: current_user.id || current_user.admin
+      @goodboy = true;
+     end
+    end
+
+    # can use this also, but i think not safe
+    #if @article.user_id == current_user.id || session[:admin]
+    # @too = true
+    #end
+    if session[:private_access] <= 0
+      render "users/Login"
+      flash[:alert] = "Reached Maximum Private Article Limit!! Do Sign or Login to continue"
+    elsif !(current_user) && !@article.public
+      session[:private_access] -= 1
+    end
+  end
+
+  # POST /articles or /articles.json
+  def create
+    if current_user
```

Comment:
User has to be logged in to create articles

Reply:
I'll change them